### PR TITLE
Cleans up turf decal and lets it be applied to atoms

### PR DESCRIPTION
**Pull request by ninjanomnom**
I forgot to allow dupes and made it use a mutable appearance to store the image instead of recreating it every time.

### CODE REVIEW

**Comment by ShizCalev** [comment]
Oh lovely, does this fix the rumtime happening with remove_decal?

**Comment by ninjanomnom** [comment]
Not directly, that's a problem with how components were changed in general. They're supposed to never be null in destroy but we're at a loss with how to do that and allow components to destroy themselves when applied incorrectly.

**Comment by ninjanomnom** [comment]
#32351 is the actual fix for that

File: code/datums/components/decal.dm [hunk 1]
```diff
@@ -0,0 +1,53 @@
+/datum/component/decal
+	dupe_mode = COMPONENT_DUPE_ALLOWED
+
+	var/cleanable
+	var/mutable_appearance/pic
+
+/datum/component/decal/Initialize(_icon, _icon_state, _dir, _cleanable=CLEAN_GOD, _color, _layer=TURF_DECAL_LAYER)
+	if(!istype(parent, /atom) || !_icon || !_icon_state)
```
Comment (Cyberboss) [inline code/datums/components/decal.dm]: i think we have isatom

File: code/datums/components/decal.dm [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+		. = COMPONENT_INCOMPATIBLE
+		CRASH("A turf decal was applied incorrectly to [parent]: icon:[_icon ? _icon : "none"] icon_state:[_icon_state ? _icon_state : "none"]")
```
Comment (Cyberboss) [inline code/datums/components/decal.dm]: parent.type

File: code/datums/components/decal.dm [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+	return ..()
+
+/datum/component/decal/OnTransfer(atom/thing)
+	remove()
+	remove(thing)
+	apply(thing)
+
+/datum/component/decal/proc/apply(atom/thing)
+	var/atom/master = thing || parent
+	master.add_overlay(pic)
```
Comment (Cyberboss) [inline code/datums/components/decal.dm]: use TRUE because it's a priority overlay

File: code/datums/components/decal.dm [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+/datum/component/decal/proc/remove(atom/thing)
+	var/atom/master = thing || parent
+	master.cut_overlay(pic)
```
Comment (Cyberboss) [inline code/datums/components/decal.dm]: ditto

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**